src/t8_vtk/t8_vtk_reader.cxx

```diff
@@ -298,13 +302,108 @@ t8_vtk_iterate_cells (vtkSmartPointer < vtkDataSet > vtkGrid,
     T8_FREE (tuples);
   }
   T8_FREE (vertices);
-  return tree_id;
+  return;
+}
+
+/**
+ * Set the partition for cmesh coming from a non-distributed vtkGrid. 
+ * 
+ * \param[in, out] cmesh  The cmesh, uncommitted, partition set after call of this function 
+ * \param[in] mpirank     The mpirank of this proc.
+ * \param[in] main_proc   The rank of the main process.
+ * \param[in] num_trees   The number of trees.
+ * \param[in] dim         The dimension of the cmesh.
+ * \param[in] comm        The communicator to use. 
+ */
+static void
+t8_vtk_cmesh_partition (t8_cmesh_t cmesh, const int mpirank,
+                        const int main_proc, t8_gloidx_t num_trees, int dim,
+                        sc_MPI_Comm comm)
+{
+  t8_gloidx_t         first_tree;
+  t8_gloidx_t         last_tree;
+  if (mpirank == main_proc) {
+    first_tree = 0;
+    last_tree = num_trees - 1;
+  }
+  /* Communicate the dimension to all processes */
+  sc_MPI_Bcast (&dim, 1, sc_MPI_INT, main_proc, comm);
+  t8_cmesh_set_dimension (cmesh, dim);
+  /* Communicate the number of trees to all processes. 
+   * TODO: This probably crashes when a vtkGrid is distributed in many 
+   * files. */
+  sc_MPI_Bcast (&num_trees, 1, T8_MPI_GLOIDX, main_proc, comm);
+
+  /* Build the partition. */
+  if (mpirank < main_proc) {
+    first_tree = 0;
+    last_tree = -1;
+    t8_geometry_c      *linear_geom = t8_geometry_linear_new (dim);
+    t8_cmesh_register_geometry (cmesh, linear_geom);
+  }
+  else if (mpirank > main_proc) {
+    first_tree = num_trees;
+    last_tree = num_trees - 1;
+    t8_geometry_c      *linear_geom = t8_geometry_linear_new (dim);
+    t8_cmesh_register_geometry (cmesh, linear_geom);
+  }
+  t8_cmesh_set_partition_range (cmesh, 3, first_tree, last_tree);
```

Comment:
It might be good style to do the following. Self-documenting code.
```suggestion
  const int set_face_knowledge = 3 /* Expect face connection of local and ghost trees. */
  t8_cmesh_set_partition_range (cmesh, set_face_knowledge , first_tree, last_tree);
```